elastic/kibana
# [Monitoring] Read from metricbeat-* for ES node_stats

Relates to #73864

This PR adapts the UI code to read from `metricbeat-*` indices for queries against ES `node_stats` type monitoring documents while preserving existing functionality. This work is based off the changes done on the beats side (https://github.com/elastic/beats/pull/19747) and must be used to test this PR. This PR should work outside of the beats PR, in that existing functionality shouldn't break at all (so it's safe to merge this to master without the beats PR going into master)

This is the first PR of several that will adapt the UI code to read from all monitoring document types, and as such, it comes with a little more code changes, mainly the changes necessary to ensure all queries start searching against `metricbeat-*` as well as handling [the fact that `type` is not aliased](https://github.com/elastic/beats/pull/19747#issuecomment-673499412).

To test:

1) Clear all existing monitoring data (to make it easier to test, you don't need to do this)
2) Pull down the [beats feature branch](https://github.com/elastic/beats/tree/feature-stack-monitoring-mb-ecs)
3) Enable the `elasticsearch-xpack` module and start Metricbeat.
4) Ensure you do not see `node_stats` in the response of this query:
```
POST .monitoring-es-*/_search
{
  "size": 0,
  "aggs": {
    "types": {
      "terms": {
        "field": "type",
        "size": 10
      }
    }
  }
}
```
5) Ensure you at least see `node_stats` in the response to this query:
```
POST metricbeat-*/_search
{
  "size": 0,
  "aggs": {
    "types": {
      "terms": {
        "field": "metricset.name",
        "size": 20
      }
    }
  }
}
```
6) Ensure the ES nodes list and ES node pages work properly
